lib/publiccloud/azure.pm

```diff
@@ -26,40 +30,53 @@ sub init {
           . $self->key_secret . ' -t ' . $self->tenantid);
 }
 
-sub filename_to_resgroup {
+sub resource_exist {
+    my ($self) = @_;
+    my $output = script_output(q(az group list --query "[?name=='openqa-upload']"));
+    return ($output ne '[]');
+}
+
+sub find_img {
     my ($self, $name) = @_;
 
+    return if (!$self->resource_exist());
+
     ($name) = $name =~ m/([^\/]+)$/;
     $name =~ s/\.xz$//;
     $name =~ s/\.vhdfixed$/.vhd/;
-    $name = $self->prefix . "-" . $name;
-    return $name;
+    my $json = script_output("az image show --resource-group " . $self->resource_group . " --name $name", 60, proceed_on_failure => 1);
+    record_info('INFO', $json);
+    eval {
+        my $image = decode_json($json);
+        return $image->{name};
+    };
 }
 
-sub find_resgroup {
-    my ($self, $name) = @_;
-
-    my $jgroups = decode_json(script_output("az group list"));
-    for my $g (@{$jgroups}) {
-        if ($name eq $g->{name}) {
-            return $name;
-        }
+sub get_storage_account_keys {
+    my ($self, %args) = @_;
+    my $output = script_output("az storage account keys list --resource-group "
```

Comment:
maybe it is task for credentials service ? just thinking loud , not sure ... 
@cfconrad WDYT ?

Reply:
I don't understand all the "magic" behind Azure and why this "keys" are needed for the later commands, but I don't think this "keys" are related to the main credentials.

Reply:
For me the question would be, is `az storage blob upload ...` possible without `az login` or even different credentials then the one belonging to our subscription.

Reply:
It makes worth the investigation. We might face this when temporal credentials are up and working. Would be nice to merge this PR asap as it´s needed by Terraform PoC.